src/bin/sol-fbp-runner/Widget.js

```diff
@@ -0,0 +1,222 @@
+(function(window) {
+
+	'use strict'
+
+	function Widget(payload, key) {
+
+		//caching a secure reference to the instance
+		var instance = this;
+		//var selectedPort;
+
+		Widget.prototype.portsCount = 0;
+
+		//unique id for the widget and other unique properties
+		this.raw = JSON.stringify(payload, null, 4);
+		this.key = key;
+		this.widgetName = payload.id; //change to 'name ?'
+		this.uid = payload.path[payload.path.length-1];
```

Comment:
ok, but you should also save `payload.path` somewhere and have some parenting/container. This will happen when you have flows that use sub flows. I'm not sure how to represent that in a nice graphical way. To the parent flow, the internal is represented as a single node with input and output ports. We can either "click to enter/expand" or show inlined with some graphical delimiter.